src/shims/windows/handle.rs

```diff
@@ -61,11 +68,16 @@ impl Handle {
             Self::Null => 0,
             Self::Pseudo(pseudo_handle) => pseudo_handle.value(),
             Self::Thread(thread) => thread.to_u32(),
+            #[expect(clippy::cast_sign_loss)]
+            Self::File(fd) => fd as u32,
+            Self::Invalid => 0x1FFFFFFF,
```

Comment:
Where does this constant come from?

Reply:
@ChrisDenton in Discord, same with the seven tag - I may be wrong, but as I understood it invalid tag is all ones so these values make that still true under miri

Reply:
Miri generally has its own handle encoding, but if "invalid handle is all ones" is a guarantee that we (have to) provide, then please
- add a comment saying that on every place that matters (the choice of 7, and right nere)
- add a test checking this

Reply:
Windows documentation doesn't have a page for INVALID_HANDLE_VALUE explicitly pinning it to -1, but several other pages do make the claim that it will be that value:

https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/get-osfhandle?view=msvc-170
https://learn.microsoft.com/en-us/dotnet/api/microsoft.win32.safehandles.safefilehandle?view=net-9.0

I'm not enough of an expert to say for sure, but it seems like the kind of thing that code might rely on even if it's strictly considered an implementation detail. I'm adding docs currently, just explaining my reasoning for this choice.

Reply:
Okay, so please add comments and a test then. I'd suggest a `#[test]` function in `handle.rs` that encodes the invalid handle and asserts it to be `u32::MAX`.